R/5-0-0summary.R

```diff
@@ -0,0 +1,34 @@
+#' Summary function
+#'
+#' @param result An object of class \code{rdlearn} returned by the
+#'   \code{\link{rdlearn}} function.
+#' @return Displays key outputs from the \code{\link{rdlearn}} function. It
+#'   provides basic information and RD causal effect estimates from
+#'   \code{\link{rdesimate}}, as well as the safe cutoffs derived by
+#'   \code{\link{rdlearn}} and the difference between them and the original
+#'   cutoffs.
+#'
+#' @export
+summary <- function(result) {
```

Comment:
`summary()` in R is kind of reserved as a S3 object. Whenever you call `summary(fit)` where `fit <- lm(y ~ x)`, `summary.lm` is called in the back. See https://adv-r.hadley.nz/s3.html#s3 for more details.

So basically, I don't recommend defining a summary function this way because it overwrites all the other "summary functions". To see this, please refer to the following simple example:

![Screenshot 2024-10-14 at 7 34 33 PM](https://github.com/user-attachments/assets/5fef091b-4815-4f45-8fc3-e1611347c637)


To avoid this, you need to define the function with a class name (`summary.rdlearn`) and treat it as a S3 object. You can find an actual example like this: https://github.com/naoki-egami/DIDdesign/blob/master/R/did-summary.R#L1-L44 or you can check [this article](https://stackoverflow.com/questions/58729497/s3-summary-method-for-an-object-in-r) or [this article](https://njtierney.github.io/r/missing%20data/rbloggers/2016/11/06/simple-s3-methods/) for more examples.

Please let me know if you have any questions!

Reply:
Thank you for your review! Are there any other specific points for improvement ?